pkg/rhelv2/rpm/rpm.go

```diff
@@ -77,33 +82,26 @@ func AddToDependencyMap(filename string, fileData analyzer.FileData, execToDeps,
 	}
 }
 
-func getCPEsUsingEmbeddedContentSets(files analyzer.Files) ([]string, error) {
-	defer metrics.ObserveListFeaturesTime(pkgFmt, "cpes", time.Now())
-
-	// Get CPEs using embedded content-set files.
-	// The files are stored in /root/buildinfo/content_manifests/ and will need to
-	// be translated using mapping file provided by Red Hat's PST team.
-	contents := getContentManifestFileContents(files)
+// getContentManifestSets returns the list of content sets defined in the content
+// manifest, if found in the files. Otherwise, `nil` or empty slice is returned.
+func getContentManifestSets(files analyzer.Files) ([]string, error) {
+	var contents []byte
+	for _, prefix := range contentManifestDirs.AsSlice() {
+		for name, file := range files.GetFilesPrefix(prefix) {
+			if strings.HasSuffix(name, ".json") {
+				// Return the first one found, as there should be only one dir per layer/node.
```

Comment:
nit: file not dir

Also, I'm not 100% sure we can guarantee it's 1 per node. An image can have more than one (I've seen up to three in the wild), and if a node is essentially a running container, then I think it's possible a node can have more than one.

Reply:
Oh, interesting. The logic was kept the same as in the previous patch. If, in fact, there can be multiple Content Sets per image/node, we need to know how to process all these content sets.

I noticed that there is some other metadata associated with each content set file, so it might be possible that when working with multiple, we need to process them differently. Or maybe concatenating them on a list. Hopefully.

I am going to keep the original data and follow up separately.

Reply:
I actually noticed that the current patch had a bug where instead of returning the first content set we would return the last one. This was fixed at e096fa77

Reply:
Maybe we should update this to allow for more than one content set per image layer (therefore node as well). That should not hurt image scanning results, and will allow for multiple content sets per node (maybe in a separate PR if we mark this with a TODO or something). I believe just appending each file's `ContentSets` into a slice is sufficient.

You bring up another point: are we sure RHCOS nodes' content_sets use the same format we are currently expecting? I'm not sure. Has this been discussed?